File: Programmers/src/Javascript/Lv0/구슬을나누는경우의수.js
function solution(balls, share) {
    return fac(balls) / (fac((balls-share)) * fac(share));
}

function fac(num) {
    let result = BigInt(1);
    
    for(let i=num; i>=2; i--)
        result *= BigInt(i);
    
    return result;
}

File: Programmers/src/Java/Lv0/구슬을 나누는 경우의 수.java
package Java.Lv0;

class Solution {
    public int solution(int balls, int share) {
        int answer = 0;
        
        answer = fac(balls, share);
        
        return answer;
    }
    
    public static int fac(int n, int m) {
        if(m==0 || n==m)
            return 1;
        else 
            return fac(n-1,m-1) + fac(n-1,m);
    }
}